<!--
  Copyright 2016 The Cartographer Authors

  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
-->

<launch>

    <!-- Arguments -->
    <arg name="config_filename" />

    <arg name="load_state_filename" default="''" />
    <arg name="load_frozen_state" default="true" />
    <arg name="save_state_filename" default="''" />

    <arg name="publish_occupancy_grid" default="false" />

    <!-- Topics -->
    <arg name="imu" default="/imu" />
    <arg name="point2" default="/velodyne_points" />

    <!-- Cartographer -->
    <node ns="cartographer" pkg="cartographer_ros" type="cartographer_node" name="cartographer_node" args="
            -configuration_filename $(arg config_filename)
            -load_state_filename $(arg load_state_filename)
            -load_frozen_state=$(arg load_frozen_state)
            -save_state_filename $(arg save_state_filename)"
            output="screen">
        <remap from="imu" to="$(arg imu)" />
        <remap from="points2" to="$(arg point2)" />
    </node>

    <node if="$(arg publish_occupancy_grid)" ns="cartographer" pkg="cartographer_ros"
        type="cartographer_occupancy_grid_node" name="cartographer_occupancy_grid_node" args="-resolution 0.05" />

</launch>
